### DISCUSSION TITLE
How do I enable websocket forwarding

### DISCUSSION THREAD

**Discussion by sandord** [dotnet/yarp#653] [Q&A] [answered]
Am I right to assume that YARP supports websocket forwarding OOTB? I'm trying to use SignalR between my REST API and my web front-end.

I've seen instructions for certain reverse proxies that requires enabling of WS upgrades but YARP doesn't seem to have such a setting. Perhaps YARP does this by default?

The web socket test seems to use a vanilla configuration so that also suggest there's nothing special to configure.

I'm using this simple configuration. There's a CORS configuration in my Startup.cs but I've come to understand that CORS is of no concern to web sockets.

    "Routes": [
      {
        "RouteId": "route1",
        "ClusterId": "cluster1",
        "CorsPolicy": "customPolicy",
        "Match": {
          "Path": "/{*remainder}"
        }
      }
    ],
    "Clusters": {
      "cluster1": {
        "Destinations": {
          "cluster1/destination1": {
            "Address": "https://localhost:52503/"
          }
        }
      }
    }

**Comment by Tratcher** [comment, marked as answer]
Correct, WebSockets should work without any additional config. Are you having trouble with it?

**Reply by sandord** [reply]
Thanks, that's good to know!

I'm currently having trouble getting authentication to work but that seems to be more of an ASP.NET thing as traffic is coming through.
